### PR TITLE
Move code from plugins install endpoint to new Plugins Library

### PR DESCRIPTION
**Pull request by ockham**
Some prep work for ~#8450~ #8482. De-duplicate all the things!

Move:
* the `Jetpack_Automatic_Install_Skin` helper class to `_inc/lib/class.jetpack-automatic-install-skin.php`
* some static helper methods from `Jetpack_JSON_API_Plugins_Install_Endpoint` to `Jetpack_Plugins` in `_inc/lib/plugins.php`.

To test: Make sure all affected endpoints still work. Note that `Jetpack_Automatic_Install_Skin` is used by a few other endpoints (grep to verify!):
* Plugin installation (from WP.org): `POST` to `/sites/%s/plugins/%s/install` (try `woocommerce`)
* Theme installation (from WP.org): `POST` to `/sites/%s/themes/%s/install` (try `twentyseventeen`) -- *see note at bottom*
* Plugin installation (via upload): `POST` to `/sites/%s/plugins/%s/new` -- use Calypso to test (`wordpress.com/plugins/upload/:site`)
* Theme installation (via upload): `POST` to `/sites/%s/themes/%s/new` -- use Calypso to test (`wordpress.com/themes/upload/:site`) 
  
Note:
* `/sites/%s/themes/%s/install` doesn't work at all, returns a 404. However, that behavior seems to be preexisting!

### CODE REVIEW

File: _inc/lib/plugins.php [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+	 *
+	 * @since 5.8.0
+	 *
+	 * @param string $slug Plugin slug.
+	 *
+	 * @return bool|WP_Error True if installation succeeded, error object otherwise.
+	 */
+	public static function install_plugin( $slug ) {
+		if ( is_multisite() && ! current_user_can( 'manage_network' ) ) {
+			return new WP_Error( 'not_allowed', 'You are not allowed to install plugins on this site.' );
```
Comment (eliorivero) [inline _inc/lib/plugins.php]: Is this error message something the user would see? If so, then it should be translatable.

File: _inc/lib/plugins.php [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+		$skin     = new Jetpack_Automatic_Install_Skin();
+		$upgrader = new Plugin_Upgrader( $skin );
+		$zip_url  = self::generate_wordpress_org_plugin_download_link( $slug );
+
+		return $upgrader->install( $zip_url );
+	}
+
+	 protected static function generate_wordpress_org_plugin_download_link( $plugin_slug ) {
+		return "https://downloads.wordpress.org/plugin/{$plugin_slug}.latest-stable.zip";
```
Comment (eliorivero) [inline _inc/lib/plugins.php]: The curly braces are not needed here since the expression is simply a string.

File: _inc/lib/plugins.php [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+			if ( self::get_slug_from_file_path( $plugin_file ) === $slug ) {
+				return $plugin_file;
+			}
+		}
+
+		return false;
+	}
+
+	protected static function get_slug_from_file_path( $plugin_file ) {
+		// Simular to get_plugin_slug() method.
```
Comment (eliorivero) [inline _inc/lib/plugins.php]: Should be "Simular".

Comment (tyxla) [reply _inc/lib/plugins.php]: You folks both probably meant "Similar" 😉

Comment (eliorivero) [reply _inc/lib/plugins.php]: ☝️

**Comment by ockham** [comment]
Ready for review!

**Comment by ockham** [comment]
Thanks @tyxla! Indentation should be fixed as of 7814897.

**Comment by ockham** [comment]
I can haz review pls @oskosk and/or @eliorivero? 😁

**Comment by oskosk** [comment]
> /sites/%s/themes/%s/install doesn't work at all, returns a 404. However, that behavior seems to be preexisting!

Tested it and it does work. It's the endpoint used by Calypso. When testing on the dev console, just don't scope the request to `v1.2`. 

`v1` or `v1.1` should work.

**Comment by ockham** [comment]
Thanks a lot for reviewing @tyxla and @oskosk!

**Comment by ockham** [comment]
@jeherve and @oskosk Would this be an okay Changelog entry?

> Move:
> * the `Jetpack_Automatic_Install_Skin` helper class to `_inc/lib/class.jetpack-automatic-install-skin.php`
> * some static helper methods from `Jetpack_JSON_API_Plugins_Install_Endpoint` to `Jetpack_Plugins` in `_inc/lib/plugins.php`.

**Comment by jeherve** [comment]
That's a bit too verbose, but don't worry about it! The label is mostly there for us, to help us build the changelog; we will come up with something as we go through all the PRs with that label at once.

**Pull request merged**